src/Lamar/ServiceRegistry.cs

```diff
@@ -15,17 +15,19 @@ public static class InstanceExtensions
         public static T Named<T>(this T instance, string name) where T : Instance
         {
             instance.Name = name;
+            instance.IsExplicitlyNamed = true;
+            
             return instance;
         }
 
         public static T Scoped<T>(this T instance) where T : Instance
         {
-            assertNotObjectInstance(instance);
+            AssertNotObjectInstance(instance);
             instance.Lifetime = ServiceLifetime.Scoped;
             return instance;
         }
 
-        private static void assertNotObjectInstance<T>(T instance) where T : Instance
+        private static void AssertNotObjectInstance<T>(T instance) where T : Instance
```

Comment:
Next time, please don't change the naming conventions. Lamar uses camel casing for private members.